src/common/Traversable.mli

```diff
@@ -0,0 +1,17 @@
+open Core_kernel
+
+module type S = sig
+  type 'a t
+
+  module A : Applicative.S
+
+  val traverse : 'a t -> f:('a -> 'b A.t) -> 'b t A.t
```

Comment:
@enetsee , is there a reason that you only implement the left-to-right traversal here? Aren't there many possible traversal orders? I see that Haskell does the same in its standard library though. I'm wondering if there's something I'm not understanding about traversals. For example, a traversal should specialize to a fold if you use a the applicative `A m _  = m` where `m` is a monoid. In that case, you at least have a left and a right fold, correct? Shouldn't you have something similar for traversals? Do you know that this traversal you implemented here is enough for our purposes?

Reply:
My understanding is that the defined traversal is sufficient, see e.g https://www.cs.ox.ac.uk/jeremy.gibbons/publications/uitbaf.pdf

There is a special case of this technique in ‘State.ml’ where I define left and right associative applicative interfaces to state and use it to define a left and right ‘map accumulate ‘ given a traversable.

Reply:
I put an example of this here: https://sketch.sh/s/FJQ9AElOGhuW5t7uxROKHS/